### DISCUSSION TITLE
scheme variable (same as host and uri)

### DISCUSSION THREAD

**Discussion by yosefy** [nginx/unit#1040] [General]
we need to redirect any uri adding trailing slash
so
```
            match: {
                uri: [
                    "*/wp-admin"
                ]
            },
            action: {
                return: 301,
                location: "http://$host$uri/"
            }
```
but would be good to make it

```
location: "$scheme://$host$uri/"
```

sometimes we don't know which protocol we use in each case

thanks

**Comment by hongzhidao** [comment]
Hi @yosefy,
Here's a draft patch, and I only tested with `http` scheme.
https://gist.github.com/hongzhidao/837c1bc55b5007ec3437805f6a76fbd9

Welcome to try it. But not sure if it's ready for the next version 1.31. We have been on it recently.
